tests/sitemaps_tests/test_generic.py

```diff
@@ -19,3 +30,23 @@ def test_generic_sitemap(self):
 </urlset>
 """ % expected
         self.assertXMLEqual(response.content.decode(), expected_content)
+
+    def test_generic_sitemap_queryset(self):
+        """Generic sitemap properly handle queryset"""
+        generic_sitemap_queryset = self.generic_sitemap.queryset.values_list(
+            "id", flat=True
+        )
+        expected_queryset = TestModel.objects.values_list("id", flat=True)
+        self.assertQuerysetEqual(
+            generic_sitemap_queryset,
+            map(repr, expected_queryset),
+            ordered=False
+        )
+
+    def test_generic_sitemap_priority_attribute(self):
+        """Generic sitemap can handle 'priority' attribute"""
+        self.assertEqual(self.generic_sitemap.priority, 0.6)
```

Comment:
I'm not sure if a separate test method for each test attribute is needed. IMO, this is making things less readable by separating the sitemap's initialization from where it's tested, especially with the unrelated `test_generic_sitemap` in the middle. There's an option to use `subTest()` if you're worried that one failure in a list of assertions will obscure other failures.